### PR TITLE
enh(resources status): Add Flapping state in RS page, resource table and status grid widget

### PR DESCRIPTION
**Pull request by Med1233**
## Description

Display the flapping state on:

- Resources status table
- Resource table widget
- Status grid widget

Capture d’écran 2024-12-02 à 16 53 01

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] 24.10.x
- [ ] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/front_src/src/Resources/Listing/ResourceListing.cypress.spec.tsx [hunk 1]
```diff
@@ -879,11 +879,73 @@ describe('Notification column', () => {
...
+      cy.findAllByRole('row')
+        .eq(1)
+        .children()
+        .first()
+        .should(
+          'have.css',
+          'background-color',
+          equals(mode, 'dark') ? 'rgb(81, 41, 128)' : 'rgb(229, 216, 243)'
+        );
+    });
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Listing/ResourceListing.cypress.spec.tsx:913]:
```suggestion
      cy.makeSnapshot();
    });
    
```

File: centreon/www/front_src/src/Resources/Listing/ResourceListing.cypress.spec.tsx [hunk 2]
```diff
@@ -879,11 +879,73 @@ describe('Notification column', () => {
...
+
+      cy.findAllByRole('row')
+        .eq(2)
+        .children()
+        .first()
+        .should(
+          'have.css',
+          'background-color',
+          equals(mode, 'dark') ? 'rgb(116, 95, 53)' : 'rgb(223, 210, 185)'
+        );
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Listing/ResourceListing.cypress.spec.tsx:929]:
```suggestion
        );
        cy.makeSnapshot();
```

**Review by Thebarda — COMMENTED** [review]
Don't forget to regenerate snapshots please

**Review by Thebarda — APPROVED** [review]

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -15152,8 +15152,8 @@ msgstr "La date de fin doit être supérieure à la date de début"
 msgid "End time"
 msgstr "Heure de fin"
 
-msgid "This Resource is flapping"
-msgstr "Cette Ressource est bagottante"
+msgid "This resource is flapping"
+msgstr "Cette ressource est bagottante"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15156]:
```suggestion
msgstr "Cette ressource est bagotante"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -19617,3 +19617,6 @@ msgstr "minute"
 
 msgid "There are errors in the form."
 msgstr "Il y a des erreurs dans le formulaire."
+
+msgid "In flapping"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19621]:
```suggestion
msgid "Flapping"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -19617,3 +19617,6 @@ msgstr "minute"
...
+msgstr "En bagotage"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:19622]:
```suggestion
msgstr "En bagotement"
```

**Review by cg-tw — APPROVED** [review]

**Pull request merged**